src/Microsoft.ML.Data/Transforms/RowShufflingTransformer.cs

```diff
@@ -651,26 +635,29 @@ protected override bool MoveNextCore()
                 {
                     // We should let the producer know it can give us more stuff.
                     // It is possible for int values to be sent beyond the
-                    // end of the sentinel, but we suppose this is irrelevant.
-                    PostAssert(_toProduce, _deadCount);
+                    // end of the Channel, but we suppose this is irrelevant.
+                    PostAssert(_toProduceChannel, _deadCount);
                     _deadCount = 0;
                 }
 
                 while (_liveCount < _poolRows && !_doneConsuming)
                 {
                     // We are under capacity. Try to get some more.
-                    int got = _toConsume.Receive();
-                    if (got == 0)
+                    var hasReadItem = _toConsumeChannel.Reader.TryRead(out int got);
```

Comment:
Previously `_toConsume.Receive()` would block the thread until something was available. But now `TryRead` will return `false` immediately if something wasn't available. In that case, we will keep spinning in the `while` loop and checking the reader constantly.

It might make sense to keep the previous behavior and block the thread until `_toConsumeChannel` has an item available, so we aren't wasting CPU cycles spinning here.